src/qibolab/platforms/abstract.py

```diff
@@ -197,16 +245,13 @@ def update(self, updates: dict):
                 # resonator_spectroscopy / resonator_spectroscopy_flux / resonator_punchout_attenuation
                 if par == "readout_frequency":
                     freq = int(value * 1e9)
-                    self.single_qubit_natives[qubit]["MZ"]["frequency"] = freq
                     self.settings["native_gates"]["single_qubit"][qubit]["MZ"]["frequency"] = freq
 
-                    if "if_frequency" in self.single_qubit_natives[qubit]["MZ"]:
-                        self.single_qubit_natives[qubit]["MZ"]["if_frequency"] = freq - self.get_lo_readout_frequency(
-                            qubit
-                        )
-                        self.settings["native_gates"]["single_qubit"][qubit]["MZ"][
-                            "if_frequency"
-                        ] = freq - self.get_lo_readout_frequency(qubit)
+                    mz = self.qubits[qubit].native_gates.MZ
+                    mz.frequency = freq
+                    if hasattr(mz, "if_frequency"):
```

Comment:
This check should be trivial, since the attribute should always be present:
https://github.com/qiboteam/qibolab/blob/1fec5b5f2d2a9153dc1f4108f55c95210d7694c9/src/qibolab/platforms/native.py#L49